src/server/scripts/Outland/HellfireCitadel/BloodFurnace/boss_broggok.cpp

```diff
@@ -118,25 +144,70 @@ struct boss_broggok : public BossAI
         }
     }
 
-    void ExecuteEvent(uint32 eventId) override
+    void UpdateAI(uint32 diff) override
     {
-        switch (eventId)
+        if (!UpdateVictim())
```

Comment:
instead of this, why not using a different EventMap for non-combat specific events?

Reply:
Doesn't really matter for me